test/test_nn.py

```diff
@@ -5126,6 +5126,21 @@ def test_pointwise_loss_target_grad_none_reduction(self):
         self.assertEqual(F.mse_loss(i, t, reduction='none').size(), t.size())
         self.assertEqual(F.l1_loss(i, t, reduction='none').size(), t.size())
 
+    def test_pointwise_loss_broadcast(self):
+        losses = {
+            'mse_loss': lambda x, y, r: F.mse_loss(x, y, reduction=r),
+            'l1_loss': lambda x, y, r: F.l1_loss(x, y, reduction=r),
+            'smooth_l1_loss': lambda x, y, r: F.smooth_l1_loss(x, y, reduction=r),
+        }
+
+        input = torch.ones(2, 1)
+        for name, fn in losses.items():
+            for requires_grad in (True, False):
+                # When target.requires_grad=True, its impl is in Python, while the other is in TH.
+                target = torch.ones(2, 10, requires_grad=requires_grad)
+                l = fn(input, target, 'none')
+                self.assertEqual(l.size(), target.size())
```

Comment:
I think one of the issues you fix is about missing backward computation. Could you also add a gradcheck here?